src/components/organisms/FolderDetail.tsx

```diff
@@ -108,6 +115,13 @@ const FolderDetail = ({ storage, folderPathname }: FolderDetailProps) => {
 
   const folderIsRoot = folderPathname === '/'
 
+  useEffect(() => {
+    const folderDoc = storage.folderMap[folderPathname]
+    if (folderDoc != null) {
+      addVisitedToHistory(folderDoc)
+    }
+  }, [addVisitedToHistory, folderPathname, storage.folderMap])
```

Comment:
We should not use an effect hook for this since the hook will be triggered when  addVisitedToHistory or storage.folderMap is changed

Reply:
How should I handle this? only check if `folder._id` changed?